app/admin/events.rb

```diff
@@ -55,5 +62,100 @@
     end
     f.actions
   end
+
+  # form controller
+  controller do
```

Comment:
nitpicking, but i would probably add some comments, like just explaining whats going on, at first glance its not super obvious to someone who hasnt looked at the code before